mbus/nats_handler.go

```diff
@@ -115,32 +120,47 @@ func (h *natsHandler) Run(handlerFunc boshhandler.Func) error {
 	}
 
 	h.runUntilInterrupted()
-
 	return nil
 }
-
 func (h *natsHandler) Start(handlerFunc boshhandler.Func) error {
 	h.RegisterAdditionalFunc(handlerFunc)
 
 	connectionInfo, err := h.getConnectionInfo()
 	if err != nil {
 		return bosherr.WrapError(err, "Getting connection info")
 	}
-
 	h.logger.Info(h.logTag, "Attempting to connect to NATS")
-
 	if net.ParseIP(connectionInfo.IP) != nil {
-		err = h.platform.DeleteARPEntryWithIP(connectionInfo.IP)
-		if err != nil {
-			h.logger.Error(h.logTag, "Cleaning ip-mac address cache for: %s", connectionInfo.IP)
-		}
+		h.arpClean()
 	}
-
 	var natsOptions = []nats.Option{
-		nats.Timeout(natsConnectionTimeout),
+		nats.DisconnectErrHandler(func(c *nats.Conn, err error) {
```

Comment:
* We don't clean the ARP cache any more on reconnect. This get lost somehow. 
* Can we block in a *handler with the `for` loop? I don't see this pattern to be used in their samples.

Reply:
I've been looking into the nats code. I think it's only blocking additional calls of error handler but I'm not quite sure yet.
The function ends up in a `sync.Cond` and looking at the logs the agent produces with the change it doesn't seem it's blocking other actions in the agent itself

Reply:
As far as the repeated flush goes, I'm not sure that is necessary more than once as the first flush already removes the old Mac from the cache and success we cannot reconnect in but sure it needs to be flushed again.

Reply:
Ok, this means you are still looking into this. Thanks for the update!